webhook_router/app.py

```diff
@@ -21,38 +98,108 @@ def handle_github_webhook() -> tuple[str, int]:
 
     Returns:
         A tuple containing an empty string and 200 status code on success or
-        a failure message and 403 status code.
+        a failure message and 4xx status code.
     """
     if secret := app.config.get("WEBHOOK_SECRET"):
-        if not (signature := request.headers.get(WEBHOOK_SIGNATURE_HEADER)):
+        is_valid, error = _validate_signature(secret)
+        if not is_valid:
+            return error, 403
+
+    is_valid, error = _validate_github_event_header()
+    if not is_valid:
+        return error, 400
+
+    try:
+        job = _parse_job()
```

Comment:
Nitpick: Just wondering if the `ParseError` should be translated to error string and return as tuples `is_valid, error` like other functions do for consistency.

Reply:
I would say that in this case `parse_job` is not a validation function, so we should not expect to have a tuple `is_valid, error, job` . This would then apply to basically all functions which could raise an exception, which seems to me to be indeed non-pythonic.